app/scripts/controllers/client/ClientChargesOverviewController.js

```diff
@@ -1,21 +1,48 @@
 (function (module) {
     mifosX.controllers = _.extend(module, {
-        ClientChargesOverviewController: function (scope, resourceFactory, location, routeParams, dateFilter) {
+        ClientChargesOverviewController: function (scope, resourceFactory, location,routeParams, dateFilter) {
             scope.clientId =  routeParams.id
+            scope.charges = [];
+            scope.actualCharges = [];
+            scope.searchText = "";
 
-            resourceFactory.clientChargesResource.getCharges({clientId: routeParams.id}, function (data) {
-                scope.charges = data;
-            });
+           scope.routeTo = function (chargeId) {
+                location.path('/viewclient/'+ scope.clientId + '/charges/'+chargeId);
+            };
 
-            resourceFactory.clientTransactionResource.getTransactions({clientId: routeParams.id}, function (data) {
-                scope.transactions = data;
-            });
+            scope.chargesPerPage =100;
 
-            scope.routeToCharge = function (chargeId) {
-                location.path('/viewclient/'+ scope.clientId + '/charges/' + chargeId);
-            };
-        }
-    });
+            scope.getClientChargeResultsPage = function (pageNumber) {
+                if(scope.searchText){
+                    var startPosition = (pageNumber - 1) * scope.chargesPerPage;
+                    scope.charges = scope.actualCharges.slice(startPosition, startPosition + scope.chargesPerPage);
+                    return;
+                }
+                var items = resourceFactory.clientChargesResource.getCharges({clientId: routeParams.id,
+                    offset: ((pageNumber - 1) * scope.chargesPerPage),
+                    limit: scope.chargesPerPage
+                }, function (data) {
+                    scope.charges = data.pageItems;
+                });
+
+            }
+
+
+            scope.initPage = function () {
```

Comment:
do you need this function ? Can you not simply all getClientChargeResultsPage (1) ?